src/components/modal/DebugModal.tsx

```diff
@@ -0,0 +1,421 @@
+// @components/modal/DebugModal.js
+import { Modal, Button, Table, Tabs, Tab, Form, InputGroup } from "react-bootstrap";
+import { use, useEffect, useMemo, useState } from "react";
+import Rest from "@rest/Rest.tsx";
+
+import dayjs from "dayjs";
+import { DISPLAY_NAME, TRANSFORMATION } from "@constants/vocabulary";
+import { GraphDBLink } from "@pages/ExecutionPage";
+
+const popModuleName = (module) => module.has_module_id?.split("/").pop();
+const popId = (exec) => exec[TRANSFORMATION].split("/").pop();
+
+const LoadingSpinner = () => (
+  <div className="mt-4 d-flex align-items-center">
+    <div className="spinner-border text-primary me-2" role="status" />
+    <span>Loading...</span>
+  </div>
+);
+
+const ErrorAlert = ({ message }) => (
+  <div className="alert alert-danger mt-3" role="alert">
+    {message}
+  </div>
+);
+
+const DebugModal = ({ show, onHide, id, name, modulesData }) => {
+  const [state, setState] = useState({
+    loading: true,
+    isComparing: false,
+    executionIdToCompare: "",
+    targetVar: "",
+    graphPatternOrigin: "",
+    graphPatternElimination: "",
+    modulesExecutions: modulesData,
+    debugData: [],
+    comparisonResult: null,
+    variableOriginResult: null,
+    tripleOriginResult: null,
+    tripleEliminationResult: null,
+    error: null,
+    activeTab: "modules",
+  });
+
+  useEffect(() => {
+    fetchExecutions();
+  }, []);
+
+  const comparableExecutions = useMemo(
+    () => state.debugData.filter((exec) => popId(exec) !== id),
+    [state.debugData, id],
+  );
+
+  const fetchExecutions = async () => {
+    try {
+      const response = await Rest.getExecutions();
+      const comparableExecutions = response.filter((exec) => popId(exec) !== id);
+      setState({ ...state, debugData: response, loading: false });
+    } catch (err) {
+      console.error("Failed to fetch executions:", err);
+      setState({ ...state, loading: false });
+    }
+  };
+
+  const compareExecutions = async (id1, id2) => {
+    try {
+      const response = await Rest.compareExecutions(id1, id2);
+      setState({ ...state, comparisonResult: response });
+      return response;
+    } catch (error) {
+      console.error("Failed to fetch modules data:", error);
+    }
+  };
+
+  const findVariableOrigin = async (id, targetVar) => {
+    try {
+      const response = await Rest.findVariableOrigin(id, targetVar);
+      setState({ ...state, variableOriginResult: response });
+      return response;
+    } catch (error) {
+      console.error("Failed to fetch modules data:", error);
+    }
+  };
+
+  const findTripleOrigin = async (id, pattern) => {
+    try {
+      const response = await Rest.findTripleOrigin(id, encodeURI(pattern));
+      setState({ ...state, tripleOriginResult: response });
+      return response;
+    } catch (error) {
+      console.error("Failed to fetch modules data:", error);
+    }
+  };
+
+  const findTripleElimination = async (id, pattern) => {
```

Comment:
Not very good practice to use just `id` ... it is not clear from context what that `id` represent ... is it `executionId`, or `moduleId` ? ... same applies for the `pattern` how about `triplePattern` ?

Please revise the file and rename vars to improve readability